src/utils/StorageUtils.ts

```diff
@@ -87,46 +87,20 @@ export default class StorageUtils {
     }
   }
 
-  public static serialize<
-    T extends
-      | FeatureGate
-      | Experiment
-      | DynamicConfig
-      | EntityNames
-      | ConfigSpecs
-      | TargetAppNames
-  >(object: T, replaceSets = false): string {
+  public static serialize(object: object): string {
     return JSON.stringify(object, (_key, value) =>
-      value instanceof Set && replaceSets ? Array.from(value) : value
+      value instanceof Set
+        ? { dataType: "Set", value: Array.from(value) }
+        : value
     );
   }
 
-  public static serializeSets<T extends EntityNames | TargetAppNames>(
-    object: T
-  ): string {
-    return this.serialize<T>(object, true);
-  }
-
-  public static deserialize<
-    T extends
-      | FeatureGate
-      | Experiment
-      | DynamicConfig
-      | EntityNames
-      | ConfigSpecs
-      | TargetAppNames
-  >(value: string, reviveSets = false): T {
+  public static deserialize<T extends object>(value: string): T {
     return JSON.parse(value, (_key, value) =>
-      value instanceof Array && reviveSets ? new Set(value) : value
+      value && value.dataType === "Set" ? new Set(value.value) : value
```

Comment:
I'm not sure that this change will be backwards compatible. Previously it was deserializing _all_ arrays as Sets, and now it only does it if they have this `dataType` field (which any existing records in the store won't have).

Are we able to flip this so that it instead keeps deserializing as a Set unless it explicitly states that it needs to be an array?

Reply:
Mm you're right, but flipping the logic won't work either since previously serialized Sets still need to be deserialized. Right now there is a mix of lists in storage that need to either be deserialized as a Set or an Array.

But I can add a check based on Assoc type. From old storage, it should be `SDK_KEYS`, `TARGET_APPS`, `TARGET_APP_TO_SDK_KEYS`, `SDK_KEY_TO_TARGET_APPS`, `TARGET_APP_TO_ENTITY_NAMES` that must be deserialized to sets